### DISCUSSION TITLE
Installed kernel modules are much larger than NV stock L4T

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1212] [Q&A] [answered]
Hi. I found out that the kernel modules installed in my image are much larger than NV stock ones.

[NVL4T.txt](https://github.com/OE4T/meta-tegra/files/11107729/NVL4T.txt)
[yocto.txt](https://github.com/OE4T/meta-tegra/files/11107730/yocto.txt)

NV L4T kernel modules only take up ~88MB space, while mine take up ~2.0GB.
The .ko files with the same name are much larger now. (6.3MB compared to 157KB)

Is there any config affecting this?

**Comment by madisongh** [comment, marked as answer]
I hadn't noticed this before, but the problem is that starting with JetPack 5, NVIDIA changed the default kernel config to set `CONFIG_MODULE_SIG` and `CONFIG_MODULE_SIG_ALL`.  The latter is causing kernel modules to get signed as part of the build process (specifically, during the `do_install` task), and so the normal debug info stripping that would happen during packaging gets skipped (because that would invalidate the signature).

There are a couple of ways this can be fixed.  I'm leaning towards disabling `CONFIG_MODULE_SIG_ALL` in our builds by default.  In the meantime, you could do this yourself via a bbappend by adding a config fragment to the SRC_URI that says `# CONFIG_MODULE_SIG_ALL is not set`.

**Reply by CHNtentes** [reply]
I tried adding a .cfg which contains `# CONFIG_MODULE_SIG_ALL is not set`, then rebuilt the kernel. However, the modules were still about 2.0GB in size. I also tried `# CONFIG_MODULE_SIG is not set`, but still didn't work.
At last, I added both lines into `arch/arm64/configs/defconfig`, still nothing changed.

**Reply by madisongh** [reply]
Well, I have this change queued up in a pending PR (see 4b9624f6d2e71be74bdf6473246444b38d9f7e8c), and it's working for me. The resulting kernel configuration is omitting that setting, and the do_package task during the kernel build is showing the modules getting stripped.  The `.ko` files are significantly smaller, as expected.

**Reply by CHNtentes** [reply]
I checked once again and it did work. The problem was that I looked in kernel recipe's image dir, not final rootfs. The .ko files got smaller when installed.
